pandas/tests/series/test_missing.py

```diff
@@ -1397,3 +1397,26 @@ def test_nonzero_warning(self):
         ser = pd.Series([1, 0, 3, 4])
         with tm.assert_produces_warning(FutureWarning):
             ser.nonzero()
+
+    def test_interpolate_non_numeric_index(self):
+        # gh 21662
+        ind = ['a', 'b', 'c', 'd']
+        df = pd.DataFrame([0, 1, np.nan, 3], index=ind)
+
+        result = df[0].interpolate()
+        expected = pd.Series([0.0, 1.0, 2.0, 3.0], name=0, index=ind)
+        assert_series_equal(result, expected)
+
+        methods = {
```

Comment:
Use the `pytest.mark.parametrize` decorator to test these methods

Reply:
@mroeschke, I have made the changes. I want to ask though, was there any particular advantage of using `pytest.mark.parameterize` here? Code readability almost remained same.

Reply:
We want to standardize our testing suite to pytest. Additionally, if there's a failure in one of the parameterized argument, it's more evident in the traceback when using this decorator.